### PR TITLE
GH-40312: [Python] Add ListView documentation to user guide

### PR DESCRIPTION
**Pull request by danepitkin**
### Rationale for this change

Add documentation of the new ListView array type.

### What changes are included in this PR?

* Update the PyArrow data types documentation page

### Are these changes tested?

Yes, built docs locally and verified.

### Are there any user-facing changes?

No, unless you count documentation updates as user-facing.
* GitHub Issue: #40312

### CODE REVIEW

**Comment by danepitkin** [comment]
image

I'm curious if you think anything else should be added to the PyArrow docs for ListView. I didn't see any other places in the user guide worth bringing it up.

File: docs/source/python/data.rst [hunk 1]
```diff
@@ -240,7 +260,7 @@ dictionaries:
 
    pa.array([{'x': 1, 'y': True}, {'z': 3.4, 'x': 4}])
 
-Struct arrays can be initialized from a sequence of Python dicts or tuples. For tuples, 
+Struct arrays can be initialized from a sequence of Python dicts or tuples. For tuples,
```
Comment (danepitkin) [inline docs/source/python/data.rst:263]: My editor auto-removed trailing whitespace found in Struct/Map documentation

**Review by AlenkaF — COMMENTED** [review]
Happy to see a documentation PR, thank you working on it! :)

I added two small comments. Can we also add a mention of list view type under **Type Metadata** section where **nested types** are mentioned?

File: docs/source/python/data.rst [hunk 2]
```diff
@@ -230,6 +230,28 @@ like lists:
...
+``pyarrow.array`` can create an alternate list type called ListView:
+
+.. ipython:: python
+
+   nested_arr = pa.array([[], None, [1, 2], [None, 1]], type=pa.list_view(pa.int64()))
+   print(nested_arr.type)
+
+ListView arrays have a different set of buffers than List arrays. The ListView array
+has both an offsets and sizes buffer, while a List array only has an offsets buffer.
+This allows for ListView arrays to specify out-of-order offsets:
```
Comment (AlenkaF) [inline docs/source/python/data.rst:245]: ❤️

**Review by AlenkaF — APPROVED** [review]
Thank you!

File: docs/source/python/data.rst (hunk 2 above)
Comment (felipecrv) [reply docs/source/python/data.rst:245]: Is it easy/possible to add a link to the spec from here?

Comment (AlenkaF) [reply docs/source/python/data.rst:245]: Sure. We could add 
```
:meth:`~pyarrow.ListViewArray`
```
to reference to the PyArrow API docs or add a reference to the existing reference object in the format/Columnar.rst section
```
:ref:`listview-layout`
```

Comment (danepitkin) [reply docs/source/python/data.rst:245]: I can add that! Great idea

Comment (danepitkin) [reply docs/source/python/data.rst:245]: https://github.com/apache/arrow/pull/40679

**Pull request merged**